### ISSUE TITLE
How do I list all triplets supported by a package?

### ISSUE THREAD

**Issue by mkckr0** [microsoft/vcpkg#13420]
`vcpkg search` only show the port name excluding the supported triplets.
```bat
> vcpkg search nlohmann-json
nlohmann-json        3.9.1            JSON for Modern C++
```

**Comment by PhoebeHui** [comment]
@mkckr0, you can use './vcpkg list xxx'.

**Comment by PhoebeHui** [comment]
I see, AFAIK, we don't have the feature now.

**Comment by BillyONeal** [comment]
@mkckr0 In general, vcpkg doesn't know in the program that a port will emit that message; vcpkg invokes `portfile.cmake` with the right settings, and the portfile is calling `vcpkg_fail_port_install`. It can decide to do that with arbitrary code. Currently we parse the Supports: field only in PR testing and continuous integration infrastructure, but `vcpkg install` should probably check the Supports field before proceeding and possibly make such data searchable.

However, the triplet .cmake is allowed to do things like `if (PORT STREQUAL "frobinate") set(VCPKG_CRT_LINKAGE "dynamic") endif()` or similar which changes how triplets are interpreted; so we might not want to do that in the general search or list commands.

@JackBoosY and I discussed Supports: out of band recently. This is my current understanding of the policy for Supports:

```
        clearly Supports -> Upstream doesn't support it
      probably Supports? -> Our port doesn't support it
probably ci.baseline.txt -> Our port tries to support it but has bugs
 clearly ci.baseline.txt -> Our port supports it but something in our CI infrastructure blocks
                               -> for example, cmcstl2, boringssl, that port where the servers are
                                  only on during business hours, etc.
                nothing! -> Everything Works
```

Jack thinks we should alter this policy like so:

![image](https://user-images.githubusercontent.com/1544943/97385006-11eb7b80-188e-11eb-9c66-7676b6499ac4.png)

That is: be more aggressive about using Supports: in these edge cases. Using Supports: more often would enable listing supported platforms or similar in the tool like @mkckr0 is asking for above.

**Comment by BillyONeal** [comment]
@ras0219 @strega-nil and I discussed this today and the group prefers the status quo (e.g. 3fd should remain in ci.baseline.txt), We only want `supports` to be used in cases where we know that the port will fail for all user machines, and for the large backlog of stuff existing in ci.baseline.txt, we do not know that is the case.

We would like to accept Jack's proposal to move ports in this edge case to "supports" rather than ci.baseline.txt.
SF F N A SA
 1  1 1 1 1

(We counted @JackBoosY as the strongly favor)

Result: No consensus to change the status quo.
